src/test/java/seedu/exercise/logic/commands/SuggestPossibleCommandTest.java

```diff
@@ -0,0 +1,74 @@
+package seedu.exercise.logic.commands;
+
+import static seedu.exercise.logic.commands.CommandTestUtil.assertCommandSuccess;
+import static seedu.exercise.model.util.DefaultPropertyManagerUtil.getDefaultPropertyManager;
+import static seedu.exercise.testutil.exercise.TypicalExercises.getTypicalExerciseBook;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Predicate;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import seedu.exercise.model.Model;
+import seedu.exercise.model.ModelManager;
+import seedu.exercise.model.ReadOnlyResourceBook;
+import seedu.exercise.model.UserPrefs;
+import seedu.exercise.model.property.Muscle;
+import seedu.exercise.model.resource.Exercise;
+
+public class SuggestPossibleCommandTest {
+
+    private Model model;
+    private Model expectedModel;
+    private Set<Muscle> targetMuscles;
+    private Map<String, String> targetCustomProperties;
+
+    @BeforeEach
+    public void setUp() {
+        model = new ModelManager(getTypicalExerciseBook(), new ReadOnlyResourceBook<>(),
+                getTypicalExerciseBook(), new ReadOnlyResourceBook<>(),
+                new UserPrefs(), getDefaultPropertyManager());
+        expectedModel = new ModelManager(model.getExerciseBookData(), new ReadOnlyResourceBook<>(),
+                getTypicalExerciseBook(), new ReadOnlyResourceBook<>(),
+                new UserPrefs(), getDefaultPropertyManager());
+        targetMuscles = new HashSet<>();
+        targetCustomProperties = new HashMap<>();
+    }
+
+    @Test
+    public void execute_suggestPossible_success() {
+        expectedModel.updateSuggestedExerciseList(getPredicate());
+        assertCommandSuccess(new SuggestPossibleCommand(targetMuscles, targetCustomProperties),
+                model, SuggestPossibleCommand.MESSAGE_SUCCESS, expectedModel);
+    }
+
+    @Test
+    public void execute_suggestPossibleMuscle_success() {
+        Muscle chest = new Muscle("Chest");
+        targetMuscles.add(chest);
+        expectedModel.updateSuggestedExerciseList(getPredicate());
+        assertCommandSuccess(new SuggestPossibleCommand(targetMuscles, targetCustomProperties),
+                model, SuggestPossibleCommand.MESSAGE_SUCCESS, expectedModel);
+    }
+
+    private Predicate<Exercise> getPredicate() {
```

Comment:
If your sets/maps are already empty, can we simplify this private `getPredicate()` method. It is extremely complex for testing and relies on multiple other components to ensure it is working

Reply:
Ok, I have updated this class. I am also considering implementing a class to handle all predicates.

Reply:
okay can do that in your next PR